controller/health.go

```diff
@@ -8,19 +8,38 @@ import (
 	"github.com/argoproj/gitops-engine/pkg/sync/ignore"
 	kubeutil "github.com/argoproj/gitops-engine/pkg/utils/kube"
 	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 
 	"github.com/argoproj/argo-cd/v2/pkg/apis/application"
 	appv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 	"github.com/argoproj/argo-cd/v2/util/lua"
 )
 
+func getLastTransitionTime(statuses []appv1.ResourceStatus, i int) metav1.Time {
+	if len(statuses) == 0 {
+		return metav1.Now()
+	}
+
+	lastTransitionTime := statuses[i].Health.LastTransitionTime
+
+	if lastTransitionTime.IsZero() {
+		lastTransitionTime = metav1.Now()
+	}
+
+	return lastTransitionTime
+}
+
 // setApplicationHealth updates the health statuses of all resources performed in the comparison
 func setApplicationHealth(resources []managedResource, statuses []appv1.ResourceStatus, resourceOverrides map[string]appv1.ResourceOverride, app *appv1.Application, persistResourceHealth bool) (*appv1.HealthStatus, error) {
 	var savedErr error
 	var errCount uint
+
+	// All statuses have the same timestamp, so we can safely get the first one
+	lastTransitionTime := getLastTransitionTime(statuses, 0)
```

Comment:
Can we directly use `app.Status.Health.LastTransitionTime` to get the value? I see the `app` object is passed down from calling `func CompareAppState` where the current PR change is using the same app obj to add the new status field value into all the resources which is then fetched using `func getLastTransitionTime`. If value of `app` is same in both funcs, I think this could simplify the logic a bit.

Reply:
You're probably right. I'll give it a go

Reply:
Yes, that works